File: src/main/java/ttd/site/generator/DateUtils.java
package ttd.site.generator;

import lombok.SneakyThrows;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

abstract class DateUtils {

	private static final DateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");

	private static final Object monitor = new Object();

	public static String formatYearMonthDay(Date date) {
		synchronized (monitor) {
			return simpleDateFormat.format(date);
		}
	}

	@SneakyThrows
	public static Date parseYearMonthDay(String pk) {
		synchronized (monitor) {
			return simpleDateFormat.parse(pk);
		}
	}

}

File: src/main/kotlin/ttd/site/generator/DateUtils.kt
package ttd.site.generator

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    private val simpleDateFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd")
    private val monitor = Object()

    @JvmStatic
    fun formatYearMonthDay(date: Date): String {
        synchronized(monitor) { return simpleDateFormat.format(date) }
    }

    @JvmStatic
    fun parseYearMonthDay(pk: String): Date {
        synchronized(monitor) { return simpleDateFormat.parse(pk) }
    }
}